src/deroff.rs

```diff
@@ -1168,6 +1155,308 @@ fn deroff_files(files: &[String]) -> io::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_text_arg() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Hello World!".into();
+    assert!(deroffer.text_arg());
+    assert_eq!(deroffer.s, " World!");
+    assert_eq!(deroffer.output.take(), "Hello");
+
+    let mut deroffer = Deroffer::new();
+    assert!(!deroffer.text_arg());
+    assert!(deroffer.s.is_empty());
+    assert!(deroffer.output.take().is_empty());
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "\t\n\t           \t\n".into();
+    assert!(!deroffer.text_arg());
+    assert_eq!(deroffer.s, "\t\n\t           \t\n");
+    assert!(deroffer.output.take().is_empty());
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = r"\r".into();
+    assert!(!deroffer.text_arg());
+    assert!(deroffer.s.is_empty());
+    assert_eq!(deroffer.output.take(), "r");
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Applebees".into();
+    assert!(deroffer.text_arg());
+    assert_eq!(deroffer.s, "");
+    assert_eq!(deroffer.output.take(), "Applebees");
+}
+
+#[test]
+fn test_font() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = r"\f(aa)lemon".into();
+    assert!(deroffer.font());
+    assert_eq!(deroffer.s, ")lemon");
+    assert!(!deroffer.font());
+    assert_eq!(deroffer.s, ")lemon");
+}
+
+#[test]
+fn test_numreq() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Hello World!".into();
+    assert!(!deroffer.numreq());
+    assert_eq!(deroffer.s, "Hello World!");
+    assert!(deroffer.output.take().is_empty());
+
+    deroffer.s = r"\w'Apple'".into();
+    assert!(deroffer.numreq());
+    assert!(deroffer.s.is_empty());
+    assert!(deroffer.output.take().is_empty());
+
+    deroffer.s = r"\w'Hello\tWorld!'".into();
+    assert!(deroffer.numreq());
+    assert_eq!(deroffer.s, "!'");
+    assert!(deroffer.output.take().is_empty());
+}
+
+#[test]
+fn test_size() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Hello World!".into();
+    assert!(!deroffer.size());
+
+    deroffer.s = r"\s10Hello World!".into();
+    assert!(deroffer.size());
+    assert_eq!(deroffer.s, "Hello World!");
+
+    deroffer.s = r"\s-11 ignore me".into();
+    assert!(deroffer.size());
+    assert_eq!(deroffer.s, " ignore me");
+}
+
+#[test]
+fn test_esc() {
+    let mut deroffer = Deroffer::new();
+    assert!(!deroffer.esc());
+
+    // This is UB, but it's the same UB as the python
```

Comment:
Which one is UB? Did you meant the previous line?

Reply:
```rust
    deroffer.s = "Hello World!".into();
    assert!(deroffer.esc());
    assert_eq!(deroffer.output.take(), "\\");
``` 
UB isn't the right word, it's just not the input it expected, It's because `esc` expects the first char to be `\`
My brain was kinda not working at that point lol